src/micromanager_gui/_plate_viewer/_analysis.py

```diff
@@ -645,6 +648,8 @@ def _extract_trace_per_position(self, p: int) -> None:
                 linear_phase = get_linear_phase(timepoints, peaks_dec_dff)
                 cubic_phase = get_cubic_phase(timepoints, peaks_dec_dff)
 
+            iei = get_iei(peaks_dec_dff, exp_time)  # s
```

Comment:
Hey @annayang26 , can you try to use the actual interval values from the metadata since we have them (`elapsed_time_list`)? Something like below...basically the `elapsed_time_list` contains the actual timestamp of each frame so I think we can use it to get the iei. I imagine that the `get_iei` method will have a  `elapsed_time_list` as argument instead of the `exposure_time`...

```suggestion
            # if the elapsed time is not available or for any reason is different from
            # the number of timepoints, set it as list of timepoints every exp_time
            if len(elapsed_time_list) != timepoints:
                elapsed_time_list = [i * exp_time for i in range(timepoints)]
            # otherwise, use the elapsed time
            else:
                iei = get_iei(peaks_dec_dff, elapsed_time_list)  # s
```

Reply:
Will do! I will change the code tomorrow.